perfkitbenchmarker/providers/azure/azure_disk.py

```diff
@@ -104,80 +109,79 @@ def __init__(self, disk_spec, vm_name, machine_type):
 
   def _Create(self):
     """Creates the disk."""
+    assert not self.is_image
 
     if self.disk_type == PREMIUM_STORAGE:
       assert FLAGS.azure_storage_type == azure_flags.PLRS
     else:
       assert FLAGS.azure_storage_type != azure_flags.PLRS
 
     with self._lock:
-      create_cmd = [AZURE_PATH,
-                    'vm',
-                    'disk',
-                    'attach-new',
-                    '--host-caching=%s' % self.host_caching,
-                    self.vm_name,
-                    str(self.disk_size)]
-      vm_util.IssueRetryableCommand(create_cmd)
-
-      if self.vm_name not in AzureDisk.num_disks:
-        AzureDisk.num_disks[self.vm_name] = 0
-      self.lun = AzureDisk.num_disks[self.vm_name]
-      AzureDisk.num_disks[self.vm_name] += 1
+      _, _, retcode = vm_util.IssueCommand(
+          [azure.AZURE_PATH, 'vm', 'disk', 'attach-new',
+           '--host-caching', self.host_caching,
+           self.vm_name, str(self.disk_size)] +
+          self.resource_group.args)
+
+      if retcode:
+        raise errors.Resource.RetryableCreationError(
+            'Error creating Azure disk.')
+
       self.created = True
 
+    @vm_util.Retry()
+    def GetName():
+      disk = self._GetDiskJSON()
+      self.name = disk['name']
+    GetName()
```

Comment:
The `GetName` function is trying to set `self.name` to the correct value. The reason it's a function is so that I can wrap it in `vm_util.Retry()`. There's no need to return it - once the variable is set, that's it.

However, your question made me realize that I can get the same retry behavior in a much cleaner way by changing `_GetDiskJSON`. Thank you! I will do that.